src/plugins/hyperbee/hyperbee.cpp

```diff
@@ -0,0 +1,1113 @@
+    {
+        fmt::print(ef_data->plugin->m_output_format, "hyperbee", drakvuf, info,
+                   keyval("EventType", fmt::Qstr("writeframe")),
+                   keyval("FrameVA", fmt::Xval(ef_data->page_va)),
+                   keyval("TrapPA", fmt::Xval(info->trap_pa)),
+                   keyval("CR3", fmt::Xval(info->regs->cr3)),
+                   keyval("GFN", fmt::Xval(info->trap->memaccess.gfn))
+                  );
+    }
+
+    drakvuf_trap* exec_Trap = create_execute_trap(info->trap->memaccess.gfn, ef_data);
+
+    //Add and activate the trap
+    drakvuf_add_trap(drakvuf, exec_Trap);
+    //store the trap to be deleted in the end.
+    ef_data->plugin->traps.emplace(exec_Trap);
+
+    //Removes this trap and frees the memory
+    ef_data->plugin->traps.erase(info->trap);
+    drakvuf_remove_trap(drakvuf, info->trap, nullptr);
+
+    PRINT_DEBUG("[HYPERBEE] Replaced Write Trap X on GFN 0x%lx with Execute trap\n", info->trap->memaccess.gfn);
+
+    return VMI_EVENT_RESPONSE_NONE;
+}
+
+/**
+ * Creates a write trap, whose pointer is returned.
+ *
+ * @param info
+ * @param ef_data
+ */
+drakvuf_trap_t* create_write_trap(drakvuf_trap_info_t* info, fault_data_struct* ef_data)
+{
+
+    auto* write_trap = (drakvuf_trap_t*) g_malloc(sizeof(drakvuf_trap_t));
+
+    //Set the type of the trap.
+    write_trap->type = MEMACCESS;
+    //Guest page-frame number to set event (as defined in events.h) (This is the level 1 translation physical address of the guest. The windows used physical address).
+    write_trap->memaccess.gfn = info->trap->memaccess.gfn;
+    write_trap->memaccess.type = POST; //Do something after sth was written
+    write_trap->memaccess.access = VMI_MEMACCESS_W; //When memory shall be written
+    write_trap->data = ef_data; //Reuse the ef_data
+
+    //Cb is the asynchronous call back https://github.com/tklengyel/drakvuf/issues/1056#issuecomment-713867399
+    write_trap->cb = write_faulted_cb;
+    write_trap->name = "write_faulted_cb";
+    return write_trap;
+}
+
+/**
+ * Creates an execute trap, whose pointer is returned.
+ *
+ * @param info
+ * @param fault_data
+ */
+drakvuf_trap_t* create_execute_trap(addr_t gfn, fault_data_struct* fault_data)
+{
+
+    auto* exec_trap = (drakvuf_trap_t*) g_malloc(sizeof(drakvuf_trap_t));
+
+    //Set the type of the trap.
+    exec_trap->type = MEMACCESS;
+    //Guest page-frame number to set event (as defined in events.h) (This is the level 1 translation physical address of the guest. The windows used physical address).
+    exec_trap->memaccess.gfn = gfn;
+    exec_trap->memaccess.type = PRE; //Do something before the access
+    exec_trap->memaccess.access = VMI_MEMACCESS_X; //When memory shall be executed
+    exec_trap->data = fault_data;
+
+    //Cb is the asynchronous call back https://github.com/tklengyel/drakvuf/issues/1056#issuecomment-713867399
+    exec_trap->cb = execute_faulted_cb;
+    exec_trap->name = "execute_faulted_cb";
+    return exec_trap;
+}
```

Comment:
I think `g_malloc` doesn't guarantee that the allocated heap memory is zeroed, so the fields of `exec_trap` that you don't explicitly assign to are treated as uninitialized. This might generate an undefined behavior. I think you should use `g_malloc0` here.

(This was caught by Valgrind, see: https://ci.drakvuf.com/job/DRAKVUF-windows7-sp1-x64/118/valgrindResult/pid=98130,0x0/ )

Reply:
Do you recommend to replace g_malloc by g_malloc0 in general? Or just at the places you pointed to?

Reply:
^ `g_try_malloc0` is recommended

Reply:
I change all mallocs to g_try_malloc0 now and added the necessary error handling

Reply:
This should be fixed now.